src/quo2/components/selectors/selectors.cljs

```diff
@@ -0,0 +1,129 @@
+     (colors/alpha colors/neutral-20 (if disabled? 0.4 1))
+     (colors/alpha colors/neutral-70 (if disabled? 0.3 1)))))
+
+(defn checkbox-prefill-view [props]
+  (let [{:keys [value onChange disabled? containerStyle]} (bean/bean props)]
+    (reagent/as-element
+     [rn/touchable-without-feedback
+      {:on-press (when (and onChange (not disabled?)) onChange)}
+      [rn/view
+       {:style        (merge
+                       (js->clj containerStyle)
+                       {:height 21
+                        :width 21
+                        :border-radius 6
+                        :background-color (colors/theme-colors
+                                           (colors/alpha colors/neutral-20 (if disabled? 0.3 1))
+                                           (colors/alpha colors/neutral-70 (if disabled? 0.3 1)))})
+        :accessibility-label (str "checkbox-" (if value "on" "off"))
+        :accessibility-role  :checkbox}
+       (when value
+         [rn/view {:style {:height 20
+                           :width 20}}
+          [icons/icon :main-icons2/check
+           {:size 20
+            :color (colors/theme-colors
+                    (colors/alpha colors/neutral-100 (if disabled? 0.3 1))
+                    (colors/alpha colors/white (if disabled? 0.3 1)))}]])]])))
+
+(defn checkbox-view [props]
+  (let [{:keys [value onChange disabled? containerStyle]} (bean/bean props)]
+    (reagent/as-element
+     [rn/touchable-without-feedback
+      {:on-press (when (and onChange (not disabled?)) onChange)}
+      [rn/view
+       {:style        (merge
+                       (js->clj containerStyle)
+                       {:height (if value 20 21)
+                        :width (if value 20 21)
+                        :border-radius 6
+                        :border-width (if value 0 1)
+                        :background-color (if value
+                                            (get-color value disabled?)
+                                            (colors/theme-colors
+                                             colors/white
+                                             (colors/alpha colors/neutral-80 0.4)))
+                        :border-color (if value :none
+                                          (get-color value disabled?))})
+
+        :accessibility-label (str "checkbox-" (if value "on" "off"))
+        :accessibility-role  :checkbox}
+       (when value
+         [rn/view {:style {:height 20
+                           :width 20}}
+          [icons/icon :main-icons2/check
+           {:size 20
+            :color (colors/alpha colors/white (if disabled? 0.3 1))}]])]])))
+
+(defn radio-view [props]
+  (let [{:keys [value onChange disabled? containerStyle]} (bean/bean props)]
+    (reagent/as-element
+     [rn/touchable-without-feedback
+      {:on-press (when (and onChange (not disabled?)) onChange)}
+      [rn/view
+       {:style        (merge
+                       (js->clj containerStyle)
```

Comment:
@J-Son89, I don't have much experience with the `bean` library, but isn't it's purpose to avoid calling functions such as `js->clj` to speedup data processing? At least that's what I got from their README. If we convert `containerStyle` to a Clojure map for merging like this function does, it seems to defeat `bean`'s purpose.

Reply:
tbh I've never seen this library before, I just copied some code from the `quo`  selectors. I can look into it but all suggestions welcome! :)

Reply:
Yeah me neither, it's an interesting library and I can easily see places in our codebase where it makes sense to use it, but if we're doing what it doesn't recommend us to do (call `js->clj`), I think it's just adding overhead to the component without any real benefit.

Reply:
removed use of bean library here